src/components/HomeScreen.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import FlightList from "./FlightList";
+import useFlightApi from "../Apis";
+
+const HomeScreen = () => {
+  const history = useNavigate();
+
+  const [flightList, setFlightList] = useState([]);
+  const [filteredFlights, setFilteredFlights] = useState([]);
+  const [errorApi, setError] = useState(null); // State to store error information
+  const [showLoading, setShowLoading] = useState(true);
+
+  const { data, loading, error } = useFlightApi(
+    "https://flight-status-mock.core.travelopia.cloud/flights"
+  );
+
+  useEffect(() => {
+    if (!loading) {
+      setFilteredFlights([...data]);
+      setFlightList([...data])
+      setError(error);
+      setShowLoading(loading);
+    }
+  }, [data, loading, error]);
+
+  const fetchData = async () => {
+    try {
+      const response = await fetch(
+        "https://flight-status-mock.core.travelopia.cloud/flights"
+      );
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+      const data = await response.json();
+      setFilteredFlights([...data]);
+      setShowLoading(false);
+    } catch (error) {
+      setError(error.message);
+      setShowLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
```

Comment:
when this happens, you should show something to the user to tell that the data is being refreshed. It can be just a label saying refreshing data, or a loading indicator. however, it shouldn't remove the exsting data from the UI

Reply:
msg added "Refreshing flight data..."